cloverage/src/cloverage/instrument.clj

```diff
@@ -31,16 +32,29 @@
       ret)
     form))
 
-(defn add-original [old new]
+(defn- add-original [old new]
   (if (iobj? new)
     (-> (propagate-line-numbers (:line (meta old)) new)
         (vary-meta merge (meta old))
         (vary-meta assoc :original old))
     new))
 
-(defn atomic-special? [sym]
+(defn- atomic-special? [sym]
   (contains? '#{quote var clojure.core/import* recur} sym))
 
+(defn- inlined-fn-call?
+  "Whether a form represents a call to a inlineable function (e.g. `clojure.core/int`) and will be replaced with the
+  inlined version."
+  [[head & args]]
+  (when (symbol? head)
+    (when-let [{:keys [inline inline-arities]} (meta (resolve head))]
+      (when inline
+        ;; some inlineable function calls such as to `+` are only replaced with the inlined version for certain arities
+        ;; -- the `:inline-arities` function in the metadata can be used to check whether the call will be inlined
+        (if inline-arities
+          (inline-arities (count args))
+          true)))))
```

Comment:
To determine whether a list is a function call that will be replaced with the inlined version we have to look at the whole form rather than just the symbol at the head of the list